### PR TITLE
 Add test programs for AFUs IBM,MEMCPY3 and IBM,AFP3

### PR DESCRIPTION
**Pull request by philippe56**

### CODE REVIEW

File: afutests/afp/ocxl_afp3.c [hunk 1]
```diff
@@ -0,0 +1,589 @@
+/*
+ * Copyright 2017 International Business Machines
+ *
```
Comment (deece) [inline afutests/afp/ocxl_afp3.c:3]: Should this be 2018?

File: afutests/afp/ocxl_afp3.c [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+		{"prefetch", no_argument, 0, 'p'},
+		{"offsetmask", required_argument, 0, 'o'},
+		{"timeout", required_argument, 0, 't'},
+		{"verbose", no_argument, &verbose,  1 },
+		{"help", no_argument, 0, 'h'},
+		{NULL, 0, 0, 0}
+	};
+
+	while ((opt = getopt_long(argc, argv, "avhc:t:", long_options, &option_index))
+	       >= 0) {
```
Comment (deece) [inline afutests/afp/ocxl_afp3.c]: Unsplit this line for readability

File: afutests/memcpy/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+===========
+
+`ocxl_memcpy` is a test program for the OpenCAPI AFU IBM,MEMCPY3 (used for
+development).
+
+Requirements
+------------
+
+The OpenCAPI card must be flashed with an IBM,MEMCPY3 AFU image.
+
```
Comment (deece) [inline afutests/memcpy/README.md:11]: Could we add a link here to help the user find where to get the image?

**Review by deece — CHANGES_REQUESTED** [review]
Please add the afutests directory to the 'precommit' target in the Makefile, then run 'make precommit'. I'm not sure if it will join statements that have been manually split (wrapped at 80 cols), so these may have to be joined manually before running precommit.

File: afutests/afp/ocxl_afp3.c (hunk 1 above)
Comment (philippe56) [reply afutests/afp/ocxl_afp3.c:3]: Correct. Fixed in commit 158b58e below.

**Review by deece — APPROVED** [review]

**Pull request merged**